### DISCUSSION TITLE
What is a gateway node?

### DISCUSSION THREAD

**Discussion by rlei-weta** [Autodesk/maya-usd#884] [Q&A] [answered]
Hello,

In the [mayaUsd ufe code](https://github.com/Autodesk/maya-usd/tree/dev/lib/mayaUsd/ufe), there are numerous references to "gateway" or "gateway node".  

What are the exact definitions of these terms?  

Are they native UFE concepts or MayaUSD UFE-specific?

**Comment by ppt-adsk** [comment, marked as answer]
The gateway node is the proxy shape node.  In theory, another implementation of USD for Maya might have another node type for this purpose, namely to be the root of the USD stage, and thus the gateway between the Maya data model and the USD data model, but in practice I'd be surprised if this ever happens...

It's not a native UFE concept; it exists only in maya-usd, for UFE needs, as you surmised.  Hope this helps.

**Reply by rlei-weta** [reply]
@ppt-adsk, gotcha!  Thanks again for the clear and concise answer!
